2-Draft-Accepted/RFCxxxx-PowerShellGet-3.0.md

```diff
@@ -0,0 +1,383 @@
+version, the first one is used.
+
+`-TrustRepository` can be used to suppress being prompted for untrusted sources.
+`-IgnoreDifferentPublisher` can be used to suppress being prompted if the publisher
+of the module is different from the currently installed version.
+`-Reinstall` can be used to re-install a module.
+
+A `-DestinationPath` parameter allows specifying the target directory instead
+of the default one.
+This will be in a different parameter set than `-Scope`.
+
+A `-NoClobber` switch will prevent installing modules that have the same cmdlets
+as a differently named module already on the system.
+
+A `-Quiet` switch will suppress progress information.
+
+`-Scope` with `AllUsers` and `CurrentUser` will work the same as it does in v2.
+Default is `CurrentUser`.
+
+`Install-Module` cmdlet will be retained for compatibility with v2.
+
+A change from v2 is that installed modules use the full semver (if used) of
+the module, so PSReadLine would be in a folder called `2.0.0-beta4` instead of
+just `2.0.0` which resolves issues of attempting to overwrite assemblies currently
+loaded on Windows.
+
+### Dependencies and version management
+
+`Install-PSResource` can accept a path to a psd1 or json file (using `-RequiredResourcesFile`),
+or a hashtable or json (using `-RequiredResources`) where the key is the module name and the
+value is either the required version specified using Nuget version range syntax or
+a hash table where `repository` is set to the URL of the repository and
+`version` contains the [Nuget version range syntax](https://docs.microsoft.com/en-us/nuget/reference/package-versioning#version-ranges-and-wildcards).
+
+```powershell
+Install-PSResource -RequiredResources @{
+  "Configuration" = "[1.3.1,2.0)"
+  "Pester"        = @{
+    version = "[4.4.2,4.7.0]"
+    repository = "https://www.powershellgallery.com"
+    credential = $cred
+  }
+}
+```
+
+The json format will be the same as if this hashtable is passed to `ConvertTo-Json`:
+
+```json
+{
+  "Pester": {
+    "version": "[4.4.2,4.7.0]",
+    "credential": null,
+    "repository": "https://www.powershellgallery.com"
+  },
+  "Configuration": "[1.3.1,2.0)"
+}
+```
+
+>[!NOTE]
+>Credentials management will be using an upcoming RFC for generalized
+>credential management.  Not sure how to make that work in json other
+>than requiring json to have credential as a json object but that means
+>password/secret is in clear text.
+
+The `repository` property can be the name of the registered repository or the URL
+to a repository.
+If no `repository` is specified, it will use the `default` repository.
+
+The older `System.Version` four part version type will be supported to retain
+compatibility with existing published modules using that format.
+
+Due to the differences between semver and `System.Version`, we may have to keep
+the two distinct.
+
+Dependent declared modules not found on the system will prompt to be installed
```

Comment:
You know, you said earlier that we want to support dependencies between scripts, modules and assemblies ... but in this section you're specifically talking about dependencies of modules during `install` ...
1. Does all of this also apply to `Save-`? 
2. Can we install scripts with the new module, or not?
3. If we don't support installing scripts, but scripts can depend on modules and assemblies ... 
    1. Currently if I `install` a script that requires modules, the modules are installed ...
    2. Currently if I `save` a script that requires modules, the modules are `saved` to the same destination folder, rather than installed -- which is OK, because if I wanted the script usable, I could have installed it, but if you take that option away, this could be very frustrating.

Reply:
`Save-` uses `Install-PSResource -DestinationPath` so the same applies.  `-DestinationPath` isn't mutually exclusive with dependencies, so it should just work.  I'll call this out explicitly.